### DISCUSSION TITLE
Can't remove value attribute for progress element

### DISCUSSION THREAD

**Discussion by wgiddens** [alpinejs/alpine#4153] [7. Bugs (archived)]
We're using a `<progress>` element to track file upload progress.  After the files is uploaded, it does a virus scan.  For the upload progress, we're setting the value from 0 to 100.   Then, during the virus scan we'd like to remove the value attribute to show progress is [indeterminate](indeterminate).   

The progress value behaves differently if it is missing than if it is set to a number.  There doesn't seem to be a way to remove the attribute altogether.  Passing a value of null or undefined will get evaluated as 0.  

Example: https://codepen.io/wgiddens/pen/vYMVgag

**Comment by ekwoka** [comment]
I'd consider this a bug. You could do it implicitely but of course that defeats some of the purpose of Alpine.

🤔 

Lets see...

So the root cause is that using normal `element.value = null` or `element.value = undefined` sets it to 0 natively.

I'll make a PR to maybe generally switch cases of `null` to be a `removeAttribute` and see if that passes tests and is a behavior we'd want in alpine...

**Comment by calebporzio** [comment]
I think I consider this an edge case, specific to the `<progress>` element. I'd suggest this workaround:

`<progress :value="value" x-effect="if (value === null) $el.removeAttribute('value')" min="0" max="100"></progress>`

If others in the future disagree, please comment and open another PR. Thanks for the effort @ekwoka
